### PR TITLE
Update to electron 17

### PR DESCRIPTION
**Pull request by Steffan153**
Well, basically, this just changes the version number. But from going from v16 to v17, `require('module').globalPaths.push(globalPath);` seemed to stop working for some reason, so I had to install [app-module-path](https://npmjs.com/app-module-path) and use it instead. It seems pretty seamless, didn't find any issues.

### CODE REVIEW

**Comment by Phylu** [comment]
Thanks a lot for your contribution.

I will create a build with the new version and test it thoroughly.

The need to change code during the update is nothing unusual. With the last electron update, I basically had to refactor the whole app.

**Comment by Phylu** [comment]
@Steffan153 Which system did you build Mailspring on (os, node version, python version, etc.)? Giving it a quick trial, I can't install the dependencies due to `node-gyp` rebuild errors.

I tried it out on macOS 12.3 with node 16 and node 17.

File: app/src/app-env.ts [hunk 1]
```diff
@@ -62,7 +62,8 @@ export default class AppEnvConstructor {
 
     // Add 'src/global/' to module search path.
     const globalPath = path.join(resourcePath, 'src', 'global');
-    require('module').globalPaths.push(globalPath);
+    // require('module').globalPaths.push(globalPath);
```
Comment (Phylu) [inline app/src/app-env.ts]: Please remove the commented out line.

**Comment by mamantoha** [comment]
Tested on my Linux with Wayland session. No issues :tada:

File: app/src/app-env.ts (hunk 1 above)
Comment (Steffan153) [reply app/src/app-env.ts]: Done

**Review by Phylu — APPROVED** [review]

**Comment by Phylu** [comment]
@Steffan153 LGTM. Thanks for the contribution. I just merged the Pull Request.

**Comment by bengotow** [comment]
Hey gang! Super exciting this was less of a pain than the move to Electron 14. I'm going to revert this merge just for a few hours to ship the 1.10.2 patch release, but then we can do some more testing on Windows + MacOS with this on master!

**Comment by Steffan153** [comment]
BTW, I forgot to mention, I tested this on Windows and it seemed to work fine. But no one's tested it on MacOS AFAIK.

**Comment by Phylu** [comment]
Just to let you know. I have been running test builds of the update for some days on my Linux (Ubuntu 21.10) and MacOS 12.3 Monterey machines. Only thing that I could not test was Windows as I am unable to build there as of now.

@Steffan153 What Windows did you build Mailspring on? If you have a combination of build tools that works natively on Windows 11, I'd be happy to get a pointer on how I can get it running :)
Last time I tried, I spent the whole weekend running from one node-gyp error into the next one without success in the end.

**Comment by Steffan153** [comment]
Win11.

I had to set `export npm_config_arch=ia32` though (well, that's `SET npm_config_arch=ia32` in Windows (`$env:npm_config_arch="ia32"` in PowerShell)). IIRC, that's all I had to do.

If you look in the task manager for the normal Mailspring, it says 32-bit, and it's set to compile to ia32 in the code somewhere. Too lazy to dig it up atm.

**Comment by Phylu** [comment]
Thanks, I got it working now using the latest node installer and some good luck. :)

**Pull request merged**